src/middlewares/error_handlers.cr

```diff
@@ -1,16 +1,44 @@
+# :nodoc:
+module ErrorHandlerHelpers
+  extend self
+
+  def messages_from(error : Exception)
+    messages = [response_data_for error]
+    while error = error.cause
+      messages << response_data_for error
+    end
+    messages
+  end
+
+  def response_data_for(error : Exception)
+    if error.responds_to? :status_code
+      # expected error handler type
+      {
+        type:        error.class.to_s,
+        message:     error.message,
+        status_code: error.status_code,
+      }
+    else
+      # any generic exception
+      {
+        type:    error.class.to_s,
+        message: error.message,
+      }
```

Comment:
Not a fan of returning a `NamedTuple`. I suggest a record, with `status_code` being nillable.
You can use the `ErrorData` from the `spec_helper` and put it there.

Reply:
Actually, I just moved the structs from spec_helper into that file, added a couple methods on them, and everything is neater. A nicer logging solution exposed itself in the process too :)

I don't like the `record` macro though, it seems silly, as it takes very little more to type out a `struct` with `property`s on it. It just seems unnecessary.